src/haz3lcore/statics/Statics.re

```diff
@@ -787,53 +810,51 @@ and uexp_to_info_map =
             kind: Abstract,
           },
         );
-      let (body, m) = go'(~ctx=ctx_body, ~mode=mode_body, body, m);
+      let (body, m) = go'(~ctx=ctx_body, ~ana=mode_body, body, m);
       add(
         ~self=Just(Forall(utpat, body.ty) |> Typ.temp),
         ~co_ctx=body.co_ctx,
         m,
       );
     | TypFun(utpat, body, _) =>
```

Comment:
I think this case is handling the case where its a hole or shadowed variable in utpat which would be much more clear if we handled it in the body of a single TypFun case